rest/server_context.go

```diff
@@ -708,75 +682,52 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 				if resyncRequired {
 					collectionsRequiringResync = append(collectionsRequiringResync, scName)
 				}
-				if err := initDataStore(dataStore, metadataIndexOption, scName); err != nil {
-					return nil, err
-				}
 			}
 		}
-		if !hasDefaultCollection {
-			if err := initDataStore(bucket.DefaultDataStore(), db.IndexesMetadataOnly, base.DefaultScopeAndCollectionName()); err != nil {
+	}
+	// no scopes, or the set of collections didn't include `_default` and we'll need to initialize it for metadata.
+	if !hasDefaultCollection {
+		ds := bucket.DefaultDataStore()
+		// No explicitly defined scopes means we'll initialize this as a usable default collection, otherwise it's for metadata only
+		if len(config.Scopes) == 0 {
+			scName := base.DefaultScopeAndCollectionName()
+			resyncRequired, err := base.InitSyncInfo(ds, config.MetadataID)
+			if err != nil {
 				return nil, err
 			}
+			if resyncRequired {
+				collectionsRequiringResync = append(collectionsRequiringResync, scName)
+			}
 		}
-	} else {
-		// no scopes configured - init the default data store
-		scName := base.DefaultScopeAndCollectionName()
-		ds := bucket.DefaultDataStore()
-		resyncRequired, err := base.InitSyncInfo(ds, config.MetadataID)
-		if err != nil {
-			return nil, err
-		}
-		if resyncRequired {
-			collectionsRequiringResync = append(collectionsRequiringResync, scName)
-		}
-		if err := initDataStore(ds, db.IndexesAll, scName); err != nil {
-			return nil, err
+		if useViews {
+			if err := db.InitializeViews(ctx, ds); err != nil {
+				return nil, err
+			}
 		}
```

Comment:
If there's no a default collection we do not need to initialize views.

Reply:
Since our tests and rosmar allows non-default collections and views, we must support this case to populate the prinicipal view in the metadata collection. Removing this makes `TestSyncGatewayStartupIndexes` fail.